File: src/falcon_core/instrument_interfaces/names/meters.py
"""A collection of knobs."""

from .dependancies import Jsonable
from .meter import Meter
from .ports import Ports


class Meters(Ports[Meter], Jsonable):
    """A collection of meters."""

    def __init__(self, meters: Meter | list[Meter]) -> None:
        """Initialize the Meters collection."""
        super().__init__(ports=meters)

File: src/instrument_interfaces/names/Meters.cpp
#include "falcon_core/instrument_interfaces/names/Meters.hpp"

namespace falcon_core::instrument_interfaces::names {

Meters::Meters() = default;
Meters::Meters(const std::vector<MeterSP>& knobs) : Ports<Meter>(knobs) {}

}  // namespace falcon_core::instrument_interfaces::names

CEREAL_REGISTER_TYPE(falcon_core::instrument_interfaces::names::Meters)
CEREAL_REGISTER_POLYMORPHIC_RELATION(
    falcon_core::generic::Song,
    falcon_core::instrument_interfaces::names::Meters)
